rfc/rfc-102/rfc-102.md

```diff
@@ -0,0 +1,116 @@
+- The RLI impl is engines compatible, for e.g, Flink can access and utilize the RLI written by Spark and vice versa;
+- The RLI is global, upserts among partitions is supported; Also support partition level RLI for large fact tables;
+- Async compaction for MDT when RLI is enabled; in writer pipeline or table services background job;
+- Smart caching of RLI;
+- Clearly set limits for the kind of write throughput supported by RLI (based on certain average response time for the RLI access, like from x0ms to x00ms) via empirical benchmarks;
+- Ability to be expanded to arbitrary secondary indexing on different columns.
+
+## Background
+Flink does not support RLI while spark does, this caused inconsistency between engines, for tables migrated from Spark to flink streaming, the index type needs to be switched to either bucket or flink_state , this caused a overhead for users in production.
+
+Another reason is for multiple partition upserts, currently the only choice is flink_state index, but the flink_state actually costs a lot of memory and can not be shared between different workloads.
+
+## Implementation
+
+The high-level ideas:
+
+- a RLI based index backend will be there to replace the flink_state index;
+- a cache of RLI would be introduced to speed the access;
+- a separate index function to write the RLI/SI payloads;
+- the MDT RLI files is written synchronously with the data table data files, the metadata is sent to the coordinator for a final commit to the MDT(after `FILES` partition is ready);
+- the MDT compaction is switched to be async and the data files compaction pipeline is reused for less take up of task slots.
+
+### The Write
+
+### The RLI Access
+In `BucketAssigner` operator, the RLI index metadata would be utilized as the index backend, the `BucketAssigner` operator will probe the RLI with the incoming record keys to figure out whether msg is update or insert or delete.
+In other words, the RLI index metadata will serve as the same role of the `flink_state` index.
+
+#### The Cache of RLI Access
+We need fast access in streaming to have high throughput(ideally per record access should be < 10ms), thus a general hotspot cache is needed. We will build a in-memory LRU cache by the active upsert records keys, the cache items will be force evictted by a configured memory threshold.
+
+We also need a memory cache for the index mappings of current checkpoint because it is not committed to Hudi table yet so invisible.
+This cache can be cleaned once the checkpoint/instant is committed to Hudi.
+
+On job restart or task failover, there is use case that the checkpoint succeeds on Flink while the instant is not committed to Hudi, for DT metadata, the pipeline will recommit
+the instant with the recovered table metadata, while for RLI access, we need to include these special inflight instants on queries, basically, we need to support reading inflight instants on MDT.
```

Comment:
> these special inflight instants

can you define such special inflight instants a bit more clearly? being able to read such inflight instants would be a general capability or a special case for flink MDT use case?